lib/int3.h

```diff
@@ -158,11 +158,11 @@ class int3
 	//returns "(x y z)" string
 	std::string toString() const
 	{
-		std::string result("(");
-		result += boost::lexical_cast<std::string>(x); result += ' ';
-		result += boost::lexical_cast<std::string>(y); result += ' ';
-		result += boost::lexical_cast<std::string>(z); result += ')';
-		return result;
+		//Performance is important here
+		char str[16] = {};

```

Comment:
as you have conversion to string anyway, it make sense to have std::string, which also has reserve method.
Optimizer of c++11 will automatically apply move semantic for returning value.

Reply:
Okay, but how do I print into that string? We're still at C++14.

Reply:
use `&str[0]`

Reply:
reserve() is useless here:
https://stackoverflow.com/questions/9521629/stdstringss-capacity-reserve-resize-functions

Reply:
yeah, indeed. Then you need to call constructor that allows passing count and character (`'\0'`).

Reply:
Or maybe it is https://www.ibm.com/docs/en/i/7.4?topic=instructions-compare-null-terminated-strings-strcmpnull

Hope this doesn't break

Reply:
`sprintf` adds trailing null, I don't think it's something what we want to add into `std::string`

Reply:
if the returned string is then compared with `==`, there can be unintended results

Reply:
No, int3 is not compared here fortunatelly.

Adding trailing `\0` won't hurt us as string is alreayd initialized wih `\0`.

Either way, I think we're overthinking it here

Reply:
Okay, this didn't work:
```
TRACE ai [7708] - Cluster Monster (110 134 0)      count: 1
TRACE ai [7708] - Cluster Monster (97 126 0)       count: 11
TRACE ai [7708] - Cluster Monster (125 40 1)       count: 1
TRACE ai [7708] - Cluster Monster (99 78 0)        count: 5
TRACE ai [7708] - Cluster Monster (78 38 1)        count: 2
TRACE ai [7708] - Cluster Monster (86 98 0)        count: 2
TRACE ai [7708] - Cluster Monster (29 8 0)         count: 3
TRACE ai [7708] - Cluster Monster (105 26 1)       count: 5
```